tests/interface/test_rdkit.py

```diff
@@ -145,15 +163,40 @@ def test_kekulization():
     )
 
 
+@pytest.mark.parametrize(
+    "values", [(False, True), (1, 2, 3), (1.0, 1.5, 2.0), ("foo", "bar", "baz")]
+)
+def test_extra_annotations(values):
+    """
+    Check if extra annotations can be added to a :class:`Mol` and recovered.
+    """
+    EXTRA_ANNOT_NAME = "extra"
+
+    ref_atoms = info.residue("ALA")
+    ref_annot = np.array(
+        list(itertools.islice(itertools.cycle(values), ref_atoms.array_length()))
+    )
+    ref_atoms.set_annotation(EXTRA_ANNOT_NAME, ref_annot)
+
+    mol = rdkit_interface.to_mol(
+        ref_atoms, include_extra_annotations=[EXTRA_ANNOT_NAME]
+    )
+    test_atoms = rdkit_interface.from_mol(mol)
+    test_annot = test_atoms.get_annotation(EXTRA_ANNOT_NAME)
+
+    assert test_annot.dtype == ref_annot.dtype
+    assert test_annot.tolist() == ref_annot.tolist()
```

Comment:
Great to see the annotation dtypes preserved! Well done for finding a way to do that :D